exasol_integration_test_docker_environment/lib/test_environment/create_ssl_certificates_task.py

```diff
@@ -0,0 +1,129 @@
+from exasol_integration_test_docker_environment.lib.data.docker_volume_info import DockerVolumeInfo
+from exasol_integration_test_docker_environment.lib.test_environment.analyze_test_container import \
+    DockerTestContainerBuild
+from exasol_integration_test_docker_environment.lib.test_environment.docker_container_copy import \
+    copy_script_to_container
+
+CERTIFICATES_MOUNT_PATH = "/certificates"
+
+
+class CreateSSLCertificatesTask(DockerBaseTask):
+    environment_name = luigi.Parameter()
+    test_container_name = luigi.Parameter()
+    docker_runtime = luigi.OptionalParameter(None, significant=False)
+    db_container_name = luigi.Parameter(significant=False)
+    network_name = luigi.Parameter()
+    reuse = luigi.BoolParameter(False, significant=False)
+    no_cleanup_after_success = luigi.BoolParameter(False, significant=False)
+    no_cleanup_after_failure = luigi.BoolParameter(False, significant=False)
+    volume_name = luigi.Parameter()
+
+    def run_task(self):
+        self.volume_info = None
+        if self.reuse:
+            self.logger.info("Try to reuse volume %s", self.volume_name)
+            try:
+                self.volume_info = self.get_volume_info(reused=True)
+            except Exception as e:
+                self.logger.warning(f"Tried to reuse volume {self.volume_name}, but got Exeception {e}. "
+                                    "Fallback to create new volume.")
+        if self.volume_info is None:
+            self.volume_info = self.create_docker_volume()
+            self.create_certificate()
+
+        self.return_object(self.volume_info)
+
+    def register_required(self):
+        self.test_container_image_future = \
+            self.register_dependency(self.create_child_task(task_class=DockerTestContainerBuild))
+
+
+    def get_volume_info(self, reused: bool) -> DockerVolumeInfo:
+        with self._get_docker_client() as docker_client:
+            volume_properties = docker_client.api.inspect_volume(self.volume_name)
+            if volume_properties['Name'] == self.volume_name:
+                return DockerVolumeInfo(volume_name=str(self.volume_name),
+                                        mount_point=volume_properties['Mountpoint'], reused=reused)
+
+    def create_docker_volume(self) -> DockerVolumeInfo:
+        self.remove_volume(self.volume_name)
+        with self._get_docker_client() as docker_client:
+            volume = docker_client.volumes.create(
+                name=self.volume_name,
+            )
+            volume_info = self.get_volume_info(reused=False)
+        return volume_info
+
+    def remove_volume(self, volume_name):
+        try:
+            with self._get_docker_client() as docker_client:
+                docker_client.volumes.get(volume_name).remove()
+                self.logger.info("Removed volume %s", volume_name)
+        except docker.errors.NotFound:
+            pass
```

Comment:
I'm not quite familiar with the project but wouldn't it be a good idea to at least leave a log of the exception handling?

Reply:
Actually, I call remove_volume() always in create_volume(), as a safe-guard. This allows re-running the integration-test-docker-environment without the need to do any manual cleanup. And it removes any artifcats in the volume.
So, adding a warning here could be a bit confusing, because the warning will always appear if running the integration-test-docker-environment for the first time.